File: 프로그래머스/Lv.2/60057. 문자열 압축/문자열 압축.java
class Solution {
    public int solution(String s) {
        int min = s.length();

        for (int i = 1; i <= s.length() / 2; i++) {
            int length = 0;

            for(int j = 0; j + i <= s.length();){
                int h = j + i;
                int count = 1;
                String seg = s.substring(j, j + i);
                
                while(
                    h + i <= s.length()
                    && seg.equals(s.substring(h, h + i))
                ){
                    h += i;
                    count++;
                } 
                if(count == 1) {
                    length += i;
                } else {
                    length += i + String.valueOf(count).length();  
                }
                j = h;
            }
            if(s.length() % i != 0) {
                length += s.length() % i; 
            }
            min = Math.min(min, length);
        }

        return min;
    }
}

File: 프로그래머스/Lv.2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    min_value = len(s);
    for i in range(1, len(s) // 2 + 1):
        length = 0
        j = 0
        while j + i <= len(s): 
            h = j + i
            count = 1
            seg = s[j:j+i]
            while h + i <= len(s) and seg == s[h:h+i]:
                h += i
                count += 1 
            if count == 1:
                length += i;
            else:
                length += i + len(str(count));  
            j = h;
        if len(s) % i != 0:
            length += len(s) % i
        min_value = min(min_value, length)
    return min_value